### ISSUE TITLE
Reduce memory footprint for gnocchi-{janitor,reporter} processes

### ISSUE THREAD

**Issue by chungg** [gnocchixyz/gnocchi#61]
this might unavoidable but when running memory_profiler

having the line "from gnocchi import service" increases memory ~5x from ~25MB to ~115MB. this seems weird as hell considering it's just initialising conf file.

**Comment by jd** [comment]
This probably imports carbonara which import pandas which imports everything (numpy etc).

**Comment by sileht** [comment]
I confirm that: 

```
etox -epy27-mysql-file mprof run python p.py
mprof: Sampling memory every 0.1s
Filename: p.py

Line #    Mem usage    Increment   Line Contents
================================================
     6     12.7 MiB      0.0 MiB   @profile
     7                             def load():
     8     13.0 MiB      0.2 MiB       import lz4.block
     9     24.4 MiB     11.5 MiB       import numpy
    10     24.4 MiB      0.0 MiB       import numpy.lib.recfunctions
    11     83.2 MiB     58.7 MiB       import pandas
    12     95.3 MiB     12.1 MiB       from scipy import ndimage
    13     95.3 MiB      0.0 MiB       import six
```

**Comment by jd** [comment]
Right, so nothing we're not aware of or we really want to fix I guess. I'm closing @chungg but feel free to reopen if you really want to fix it.

**Comment by chungg** [comment]
hmm. i'd like to minimise memory... i think it's possible to make janitor and reporter workers to be less but we'll have way more processing workers which i'm not sure we can minimise if pandas is the main reason. i'll leave it closed unless you folks have ideas.

**Comment by jd** [comment]
I'll reopen because I think it's possible and it's a good idea. I did not think about the impact on reporter/janitor.

We might need to split service.py and write some tests to make we don't break that again later. It's easy to be greedy on import.

**Comment by chungg** [comment]
just looking at this more. the switch to numpy has been awesome. the remaining pandas usage i see are for 2 things, either: for processing timespan input or for computing cross metric computations. both of these are used solely for API so it seems we could avoid pulling in pandas for metricd if we correctly separate the pandas code to API.

**Comment by jd** [comment]
Yes, and both those usage should be removed. I'll probably need @sileht help to remove the cross metric computing (since he wrote it), but the time parsing should be not too hard to extract from pandas.

**Comment by chungg** [comment]
i'm wondering if we should add a new API service that does analysis or something. there's stuff like moving/rolling average/max/min/etc... which is useful but may need pandas (or maybe not)

**Comment by chungg** [comment]
just some random notes on places we can potential reduce memory

`6.6 MiB       from oslo_middleware import cors` is loaded by non-api workers from gnocchi.service
`15.4 MiB       from tooz import coordination` is loaded by non-processing metricd workers from gnocchi.utils

**Comment by jd** [comment]
@chungg I think you just fixed that with 3d5c5c5b93ef0868a3122afc09049c97bdbeea82

is there anything else we can do or should we close that bug?

**Comment by chungg** [comment]
we can close it. i think the memory usage is high still but we can look more into later.
